backend/danswer/background/celery/celery_app.py

```diff
@@ -635,6 +691,89 @@ def vespa_metadata_sync_task(self: Task, document_id: str) -> bool:
     return True
 
 
+@celery_app.task(
+    name="document_by_cc_pair_cleanup_task",
+    bind=True,
+    soft_time_limit=45,
+    time_limit=60,
+    max_retries=3,
+)
+def document_by_cc_pair_cleanup_task(
+    self: Task, document_id: str, connector_id: int, credential_id: int
+) -> bool:
+    task_logger.info(f"document_id={document_id}")
+
+    try:
+        with Session(get_sqlalchemy_engine()) as db_session:
+            curr_ind_name, sec_ind_name = get_both_index_names(db_session)
+            document_index = get_default_document_index(
+                primary_index_name=curr_ind_name, secondary_index_name=sec_ind_name
+            )
+
+            count = get_document_connector_count(db_session, document_id)
+            if count == 1:
+                # count == 1 means this is the only remaining cc_pair reference to the doc
+                # delete it
+                document_index.delete(doc_ids=[document_id])
+                delete_documents_complete__no_commit(
+                    db_session=db_session,
+                    document_ids=[document_id],
+                )
+            elif count > 1:
+                doc = get_document(document_id, db_session)
+                if not doc:
+                    return False
+
+                doc_access = get_access_for_document(
```

Comment:
can we add a comment here about why these will be correct e.g. it will have the correct access since we don't consider deleting cc pairs in `get_access_for_document` / `fetch_document_sets_for_document`)?

I got confused myself for a second, and thought there was a bug here